Modules/Core/Common/include/itkExtractImageFilter.hxx

```diff
@@ -75,7 +75,7 @@ ExtractImageFilter<TInputImage, TOutputImage>::SetExtractionRegion(InputImageReg
    */
   for (unsigned int i = 0; i < InputImageDimension; ++i)
   {
-    if (inputSize[i])
+    if (inputSize[i] and nonzeroSizeCount < OutputImageDimension)
```

Comment:
Looking at https://github.com/InsightSoftwareConsortium/ITK/blob/9d1a93dfb8edb10bce000f9695f9e626b3ed2a26/Modules/Core/Common/include/itkExtractImageFilter.hxx#L86-L92

It appears that the originally intended behavior was to throw an exception when nonzeroSizeCount gets too big. However, your proposed solution would "silently" prevent nonzeroSizeCount from getting too big. That's different behavior! I don't know which behavior is preferred 🤷  

BTW (nitpick alarm 😸): Most C++ programmers use `&&`, instead of `and`!

Reply:
VS [chokes](https://open.cdash.org/viewBuildError.php?buildid=9565895) on `and`:
D:\a\1\s\Modules\Core\Common\include\itkExtractImageFilter.hxx(78): error C2065: 'and': undeclared identifier

Reply:
my mistake.  jumping between python an C++.  I was doing this in the background while waiting for my python data analysis to complete.  Sorry.

Reply:
@N-Dekker THANKS!
Fixed to 
```c++
  for (unsigned int i = 0; i < InputImageDimension && nonzeroSizeCount < OutputImageDimension; ++i)
  {
    if (inputSize[i])
    {
      outputSize[nonzeroSizeCount] = inputSize[i];
      outputIndex[nonzeroSizeCount] = extractRegion.GetIndex()[i];
      ++nonzeroSizeCount;
    }
  }
```

Reply:
You're welcome, but the point I was trying to make was that the original code would throw an exception if nonzeroSizeCount != OutputImageDimension. Wasn't that the intended behavior when nonzeroSizeCount gets too big?

Reply:
So I would suggest, instead:
```cpp
  for (unsigned int i = 0; i < InputImageDimension; ++i)
  {
    if (inputSize[i])
    {
      if (nonzeroSizeCount < OutputImageDimension)
      {
        outputSize[nonzeroSizeCount] = inputSize[i];
        outputIndex[nonzeroSizeCount] = extractRegion.GetIndex()[i];
      }
      ++nonzeroSizeCount;
    }
  }
```
In order to get into the `if` clause of `if (nonzeroSizeCount != OutputImageDimension)`, to throw an exception, when nonzeroSizeCount gets too big:

https://github.com/InsightSoftwareConsortium/ITK/blob/9d1a93dfb8edb10bce000f9695f9e626b3ed2a26/Modules/Core/Common/include/itkExtractImageFilter.hxx#L86-L92

Reply:
@N-Dekker I looked at this again, and your solution is exactly what is needed. 

Thank you for your careful inspection.  Update to PR coming soon.